### ISSUE TITLE
Preview script arg1=FILE_PATH dereferencing breaks git-annex usage

### ISSUE THREAD

**Issue by amerlyq** [ranger/ranger#2276]
#### Current Behavior
All files are dereferenced before being passed to `scope.sh`.
```bash
FILE_PATH=/tmp/repo/.git/annex/objects/bcc/SHA256-s31343960--0de3936ae3afeda2f88e850e7eea7ef89caf1504c7fe5cfdd353df0f5dbcf2fe/SHA256-s31343960--0de3936ae3afeda2f88e850e7eea7ef89caf1504c7fe5cfdd353df0f5dbcf2fe
```

#### Expected Behavior
Script `scope.sh` must get the same path, which is shown in ranger under cursor:
```bash
FILE_PATH=/tmp/repo/large_image.jpg
```

#### Context
I use git-annex and keep large blobs (images, pdfs, archives, etc.) together with project sources under single git repo.
The way git-annex works is by replacing original files by symlinks to extensionless SHA files (as was intentionally configured) stored in externally synced storage.

As a result, lack of extension breaks all preview.
Moreover we lose information on original directory where symlink was placed -- so `scope.sh` can't extract context (e.g. read some metadata from `.previewrc` in same directory).

I know about security concerns, which forced chromium to always use realpath on all local files https://github.com/qutebrowser/qutebrowser/issues/5858, but it's not the case for ranger, as the `scope.sh` script isn't strictly external file and more like part of internal machineria, and so it must have access to some of internal state of ranger to work "as expected".

#### Possible Solutions
Here we use "realpath" for both image cache store (which is good and must be left as is) and for passing args to `scope.sh`.
I propose to replace `args=[..., path, ...]` by  `args=[..., fobj.path, ...]` to prevent path resolution (maybe better use another property?).
```python
      # ./ranger/core/actions.py:1036
      def get_preview(self, fobj, width, height):
          # pylint: disable=too-many-return-statements,too-many-statements
          pager = self.ui.get_pager()
          path = fobj.realpath
          ...
        loadable = CommandLoader(
            args=[self.settings.preview_script, path, str(width), str(height),
```
Does anybody have any objections to such change?

**Comment by toonn** [comment]
We consider `scope.sh` 100% a user-configured file, there's nothing internal about it and it shouldn't be deeply linked to ranger. You could replace it with any script in any language or executable, it just has to follow the convention re argument passing. (This is just FYI.)

Thanks for clearly explaining why you want this. It sounds very reasonable to me and [I don't foresee any issues](https://xkcd.com/1172/) except for the other way around where people use symlinks that are hashes or otherwise short forms.

@vifon, do you see any good reasons for only passing realpaths to scope?

**Comment by vifon** [comment]
Not really, it's just how we've done it. But we can add an "unreal" path too.

@amerlyq, mostly out of curiosity, what was your intent when you configured Annex to skip the extensions?

**Comment by amerlyq** [comment]
> But we can add an "unreal" path too.

Actually, we already pass so much arguments on cmdline, that using env vars for that starts looking "not too bad" :D
On the other hand we don't really need passing realpath to `scope.sh`, because simple call to `realpath` will resove it if needed inside script itself.

> what was your intent when you configured Annex to skip the extensions?

* default SHA256E backend will create duplicates for files ./aaa.jpg, ./aaa.JPG and ./aaa.56.jpg, I didn't liked that
* I had too many legacy files from mixed windows / linux reserve copies which I needed to merge at one point -- all of them contained mix of extension case
* some files simply had wrong extensions and I didn't wish to re-add them again after renaming in future
* some files didn't have extension at all, and I knew them were duplicates.
* manual conversion of files on NAS (with slow CPU -- i.e. circumventing hashing) was easier to be automated w/o extensions
* list of "broken" software which always require extension was small and I didn't care about any of them. After all it was long before chromium fixed CVE and started using realpath everywhere
* idealism: symlinks on linux must be treated as proper files themselves in their original context, and not resolved immediately, loosing all their context. So I intentionally seeked "broken" programs to nag :)

Basically, I think now after renormalizing everything I would stick with SHA256e to spend less time on fighting windmills and chasing pipe dreams. But still, by using realpath we lose original context of symlink i.e. you won't know which files were beside it in the same dir, so adding extensions to git-annex now isn't complete panacea to the problem.

**Comment by vifon** [comment]
> Actually, we already pass so much arguments on cmdline, that using env vars for that starts looking "not too bad" :D
On the other hand we don't really need passing realpath to scope.sh, because simple call to realpath will resove it if needed inside script itself.

+1 on both.

> [a long list of points I'm not going to copy-paste]

I see now, makes sense. And yes, I agree with the last point in the context of ranger too, we shouldn't need to dereference the links at all in `scope`. Especially since we don't do that elsewhere, as I use ranger with Git Annex a lot and I would notice.
